src/socket.cpp

```diff
@@ -2521,7 +2521,15 @@ int open_connections()
         struct addrinfo  hints = {AI_PASSIVE, AF_UNSPEC,};
 
         if (!strlen(local_ip)) {
-            local_host = (char *)hostname;
+            if (sendMode == MODE_CLIENT) {
+                if (remote_sockaddr.ss_family == AF_INET) {
+                    local_host = strdup("0.0.0.0");
+                } else {
+                    local_host = strdup("::");
```

Comment:
`local_host` can actually be a `const char*`, so you can change this to `local_host = "::";` and drop the casts. Otherwise this will leak memory.